bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleRegistryImpl.java

```diff
@@ -89,77 +94,177 @@
  * @author Benedikt Niehues - added events for rules
  */
 public class RuleRegistryImpl extends AbstractRegistry<Rule, String, RuleProvider>
-        implements RuleRegistry, StatusInfoCallback {
+        implements RuleRegistry, StatusInfoCallback, ManagedService {
```

Comment:
Ok. Now I got your initial idea and yes it makes sense to make use of the component configuration(of RuleRegistry) rather than impl ManagedService.